### PR TITLE
DOC: GH5456 Adding workaround info on NA / NaT handling for groupby

### PR DESCRIPTION
**Pull request by aamnv**
- [x] closes #5456
- [NA] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [NA] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [NA] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Took a stab at this by parsing through the initial issue and related topics - let me know if the intent was different, happy to edit.

### CODE REVIEW

File: doc/source/user_guide/groupby.rst [hunk 1]
```diff
@@ -1268,6 +1268,11 @@ automatically excluded. In other words, there will never be an "NA group" or
 generally discarding the NA group anyway (and supporting it was an
 implementation headache).
 
+.. note::
+   If you need to include NaN or NaT values in your grouping, you can workaround the automatic exclusion by replacing
+   the NaN / NaT values with a placeholder string. For example, you can use ``df.fillna("default", inplace=True)`` to
```
Comment (mroeschke) [inline doc/source/user_guide/groupby.rst:1273]: Thanks for the PR! 

1.  Could you demonstrate this by including a `.. ipython:: python` example (like above)
2. In the example, could you exclude `inplace=True`? We are trying to discourage that keyword generally

Comment (aamnv) [reply doc/source/user_guide/groupby.rst:1273]: Can do on both!

Do you think it makes sense to pull it out of a note and just include it in line with the rest of the text if we're including an ipython?

Comment (mroeschke) [reply doc/source/user_guide/groupby.rst:1273]: Yeah sure that sounds reasonable

**Review by phofl — COMMENTED** [review]
I might be missing something here, but these days you can simply set ``dropna=False``, which would include the NaNs

```


df = pd.DataFrame({"a": [1, np.nan, 2], "b": 3})

print(df.groupby("a", dropna=False).sum())
```

returns 

```
     b
a     
1.0  3
2.0  3
NaN  3
```

**Comment by mroeschke** [comment]
Good point @phofl. Looks we also have an example in the groupby docstring demonstrating this behavior so I don't think this workaround explanation is needed. 

Thanks for the PR @aamnv but going to close since I think this old issue is not actionable anymore, but feel free to address any of the other `good first issue` issues!

**Pull request closed without merging**